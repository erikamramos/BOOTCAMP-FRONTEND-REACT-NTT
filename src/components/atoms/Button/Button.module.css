.button {
  padding: var(--padding-normal);
  box-shadow: var(--shadow-inset);
  border-radius: var(--radius-large);
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border: none;
  font-family: var(--font-primary);
  font-weight: 600;
  font-size: 0.8rem;
  gap: 5px;
  transition: transform var(--transition-duration);
}

.button:hover {
  transform: scale(1.05);
}

.button--primary {
  background-color: var(--color-primary);
  color: var(--color-secondary);
}

.button--primary:hover {
  color: var(--color-secondary);
  background-color: var(--color-primary);
}

.button--secondary {
  background-color: var(--color-button-bg);
  color: var(--color-primary);
}

.button--secondary:hover {
  color: var(--color-secondary);
  background-color: var(--color-primary);
}

.button--disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.button--auto {
  width: auto;
}

.button--block {
  width: 100%;
}
