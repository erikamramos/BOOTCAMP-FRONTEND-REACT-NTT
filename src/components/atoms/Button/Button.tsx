import { FC, PropsWithChildren } from 'react';
import styles from './Button.module.css';
import { Icon } from '../Icon/Icon';
import { ButtonConfig } from '../../../config/constants/config';

type ButtonProps = {
  onClick?: () => void;
  variant?: 'primary' | 'secondary';
  block?: boolean;
  disabled?: boolean;
  icon?: string | null;
  className?: string;
};

export const Button: FC<PropsWithChildren<ButtonProps>> = ({
  children,
  onClick,
  variant = 'primary',
  block = false,
  disabled = false,
  icon,
  className,
}) => {
  return (
    <button
      className={`
        ${className}
        ${styles.button} 
        ${variant === 'primary' ? styles['button--primary'] : styles['button--secondary']}
        ${block ? styles['button--block'] : styles['button--auto']}
        `}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
      {icon && (
        <Icon
          name={icon}
          color={variant === 'primary' ? ButtonConfig.PRIMARY_COLOR : ButtonConfig.SECONDARY_COLOR}
          size={ButtonConfig.DEFAULT_SIZE}
        />
      )}
    </button>
  );
};
