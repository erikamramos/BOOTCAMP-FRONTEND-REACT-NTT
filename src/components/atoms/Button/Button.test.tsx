import { render, screen } from '@testing-library/react';
import { Button } from '@/components/atoms/Button/Button';

const renderComponent = (props = { children: 'Click me' }) => {
  return render(<Button {...props}>{props.children}</Button>);
};

describe('Button Component', () => {
  it('renders component', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByText(/click me/i)).toBeInTheDocument();
  });

  it('renders with different children', () => {
    renderComponent({ children: 'Submit' });
    expect(screen.getByText(/submit/i)).toBeInTheDocument();
  });
});
