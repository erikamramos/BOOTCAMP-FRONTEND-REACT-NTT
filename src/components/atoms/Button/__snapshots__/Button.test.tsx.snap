// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders component 1`] = `
<div>
  <button
    class="
        undefined
        button 
        button--primary
        button--auto
        "
  >
    Click me
  </button>
</div>
`;
